app/dashboard/models.py

```diff
@@ -2400,6 +2430,136 @@ def is_staff(self):
         """
         return self.user.is_staff if self.user else False
 
+    def calc_activity_level(self):
+        """Determines the activity level of a user
+
+        Returns:
+            str: High, Low, Medium, or New
+
+        """
+        high_threshold = 15
+        med_threshold = 15
+        new_threshold_days = 7
+
+        if self.created_on > (timezone.now() - timezone.timedelta(days=new_threshold_days)):
+            return "New"
+
+        visits = self.actions.filter(action='Visit')
+        visits_last_month = visits.filter(created_on__gt=timezone.now() - timezone.timedelta(days=30)).count()
+
+        if visits_last_month > high_threshold:
+            return "High"
+        if visits_last_month > med_threshold:
+            return "Medium"
+        if visits_last_month > med_threshold:
+            return "Low"
+
+
+    def calc_longest_streak(self):
+        """ Determines the longest streak, in workdays, of this user
+
+        Returns:
+            int: a number of weekdays
+
+        """
+
+        # setup
+        action_dates = self.actions.all().values_list('created_on', flat=True)
+        start_date = timezone.datetime(self.created_on.year, self.created_on.month, self.created_on.day)
+        end_date = timezone.datetime(timezone.now().year, timezone.now().month, timezone.now().day)
+
+        # loop setup
+        iterdate = start_date
+        max_streak = 0
+        this_streak = 0
+        while iterdate < end_date:
+            # housekeeping
+            last_iterdate = start_date
+            iterdate += timezone.timedelta(days=1)
+
+            is_weekday = iterdate.weekday() < 5
+            if not is_weekday:
```

Comment:
shouldn't weekends be counted in the streak as well? 

also is the `if not is_weekday` grabbing weekends here? if its 5 or greater `is_weekday` is false and the `if not` will render `true`, and it'll continue, no?

Reply:
derp it should only count weekdays..  will update

 i dont think weekends should count in the streak mechanic.. we dont want to build a culture that encourages burnout